relay-server/src/utils/unreal.rs

```diff
@@ -201,7 +201,8 @@ fn merge_unreal_context(event: &mut Event, context: Unreal4Context) {
 
     if let Some(login_id) = &runtime_props.login_id {
         let id = event.user.get_or_insert_with(User::default).id.value_mut();
-        if id.is_none() {
+
+        if id.as_ref().map_or(true, |s| s.is_empty()) {
```

Comment:
I learned of it too late, but this seems a good usecase for `Option::is_none_or`.
```suggestion
        if id.as_ref().is_none_or(|s| s.is_empty()) {

```

Reply:
Huh, I remember searching for this a while ago and being annoyed that it doesn't exist, but they added it with 1.82 now!